src/shader_recompiler/frontend/translate/vector_alu.cpp

```diff
@@ -595,4 +604,27 @@ void Translator::V_CMP_CLASS_F32(const GcnInst& inst) {
     }
 }
 
+void Translator::V_MAD_U32(const GcnInst& inst) {
+    const IR::U32 src0{GetSrc(inst.src[0])};
+    const IR::U32 src1{GetSrc(inst.src[1])};
+    const IR::U32 src2{GetSrc(inst.src[2])};
+    // TODO
+    const IR::U32 result = ir.IAdd(ir.IMul(src0, src1), src2);
```

Comment:
This is incomplete from my branch, to get this in main you will need to add proper bitfield extracts as the ISA documentation specifies for V_MAD_U32_U24 (also change the name to that)

Reply:
Now the V_MAD_U32 function is performing a 24-bit extraction. However, I'm a little confused about whether this function should handle 32-bit operations instead.

Additionally, the V_MAD_U32_U24 function is calling V_MAD_I32_I24(inst, false). Should we consider changing this function as well?
updated code: https://github.com/shadps4-emu/shadPS4/pull/333/files#diff-9a6c2e2027c03231e88aaaab30908baecae202661839f35c31a777fec2500c7aR606